### ISSUE TITLE
Internal error at src/full/Agda/Syntax/Concrete/Definitions.hs:904

### ISSUE THREAD

**Issue by wenkokke** [agda/agda#4881]
You can trigger the following internal error:
```
An internal error has occurred. Please report this as a bug.
Location of the error: src/full/Agda/Syntax/Concrete/Definitions.hs:904
```
Minimal test case:
```
_ : ?
```

**Comment by gallais** [comment]
With 2.6.1 I get:

```agda
?0 : _0
Sort _0  [ at PATH/MissingDef.agda:1,5-9 ]

———— Errors ————————————————————————————————————————————————
PATH/MissingDef.agda:1,1-9
The following names are declared but not accompanied by a
definition: _
```

**Comment by wenkokke** [comment]
Yeah, it's a problem in the HEAD version of Agda only.

**Comment by uncle-betty** [comment]
Perhaps this is a consequence of the fix for #4157?

I was able to gather the following additional data.

For `NoName` names, `addLoneSig` now generates a fresh `NameId` when adding a new signature to the `LoneSigs` map. However, the map key uses the old `NameId`, so that there's a mismatch between the map key's `NameId` and the map value's `NameId`. This is intentional, according to the comment on `loneSigName`.

Here's what happens inside `niceDeclarations`in the above MWE:

* The single `Declaration` in the list `ds` uses `NameId 0 0`.
* The `nice` function adds the `Declaration` from `ds` to the `LoneSigs` map via `addLoneSig`. The fresh `NameId` created by `addLoneSig` (`NameId 1 1`) is also used in the `NiceDeclaration` list produced by `nice`, which is assigned to `nds`.
* `replaceSigs` is applied to the `LoneSigs` map (key `NameId 0 0`, value `NameId 1 1`) and to `nds` (`NameId 1 1`).
* This gets us into the `d:ds` case in `replaceSigs`.
* `Map.updateLookupWithKey` uses the `NameId 1 1` taken from `nds` for the lookup in the `LoneSigs` map, which yields `Nothing` as the winning key would have been `NameId 0 0`.
* This `Nothing` leads to recursion with a now empty `nds` and unchanged (i.e., non-empty) `ps`. This triggers the `__IMPOSSIBLE__`.

**Comment by uncle-betty** [comment]
Whoops, had I known that the commits to my fork show up here, I wouldn't have amended and force-pushed. Anyway. #4901 now is my proposal for fixing this.
